src/hooks/app-data-provider/useAppDataProvider.tsx

```diff
@@ -92,52 +93,53 @@ export const AppDataProvider: React.FC = ({ children }) => {
     userReserves,
     userEmodeCategoryId,
     eModes,
-    ghoReserveData,
-    ghoUserData,
-    ghoReserveDataFetched,
     formattedPoolReserves,
     userSummary,
-    displayGho,
   ] = useRootStore((state) => [
     selectCurrentReserves(state),
     selectCurrentBaseCurrencyData(state),
     selectCurrentUserReserves(state),
     selectCurrentUserEmodeCategoryId(state),
     selectEmodes(state),
-    state.ghoReserveData,
-    state.ghoUserData,
-    state.ghoReserveDataFetched,
     selectFormattedReserves(state, currentTimestamp),
     selectUserSummaryAndIncentives(state, currentTimestamp),
-    state.displayGho,
   ]);
 
-  const formattedGhoReserveData: FormattedGhoReserveData = formatGhoReserveData({
-    ghoReserveData,
-  });
-  const formattedGhoUserData: FormattedGhoUserData = formatGhoUserData({
-    ghoReserveData,
-    ghoUserData,
-    currentTimestamp,
-  });
+  const currentMarketData = useRootStore((state) => state.currentMarketData);
+
+  const { data: formattedGhoUserData, isLoading: isGhoUserDataLoading } =
+    useUserGhoPoolFormattedReserve(currentMarketData);
+  const { data: formattedGhoReserveData, isLoading: ghoReserveDataLoading } =
+    useGhoPoolFormattedReserve(currentMarketData);
+
+  const formattedGhoReserveDataWithDefault = formattedGhoReserveData || {
+    aaveFacilitatorRemainingCapacity: 0,
```

Comment:
This sadly is a workaround, i dont think is good idea to add `initialData` or `placeholderData` here because they turn the query instantly into `success` state and loading states are false always -> https://tkdodo.eu/blog/placeholder-and-initial-data-in-react-query#similarities
We do need the default state for the data because components that consume `useAppDataProvider` doesnt check if the data is exists or not, so thats why this workaround
In the future the idea is to completly delete `useAppDataProvider` and make the components consume directly the hook, doing the typechecks there